File: app/src/main/java/ru/org/adons/cblock/ui/base/ViewScope.java
package ru.org.adons.cblock.ui.base;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import javax.inject.Scope;

import ru.org.adons.cblock.app.ApplicationComponent;

/**
 * Scope for all sub-component of {@link ApplicationComponent}
 */
@Scope
@Retention(RetentionPolicy.RUNTIME)
public @interface ViewScope {
}

File: app/src/main/java/mobile/addons/cblock/scope/ViewScope.kt
package mobile.addons.cblock.scope

import mobile.addons.cblock.app.ApplicationComponent
import javax.inject.Scope

/**
 * Scope for all sub-component of [ApplicationComponent]
 */
@Scope
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
annotation class ViewScope
